pkg/util/gpg.go

```diff
@@ -79,14 +87,18 @@ func verifyGPGSignature(keyring *openpgp.EntityList, message []byte, signature *
 		}
 	}
 
-	//add the downloaded keys to the existing keyring
-	el, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(publicKeyBytes))
-	if err != nil {
-		return err
+	//add the downloaded keys to the existing key ring
+	if len(publicKeyBytes) != 0 {
+		el, err := openpgp.ReadArmoredKeyRing(bytes.NewReader(publicKeyBytes))
+		if err != nil {
+			return err
+		}
+		keyring.Mux.Lock()
+		keyring.EntityList = append(keyring.EntityList, el...)
+		keyring.Mux.Unlock()
```

Comment:
When you guard a variable with a mutex, you need to guard *all* accesses to it, including read access, e.g. calls to `keyring.EntityList.KeysById()`. The recommended way to do this is with [sync.RWMutex](https://golang.org/pkg/sync/#RWMutex).